File: src/main/scala-graphql/io/github/graphql/model/TopicSuggestionDeclineReasonTO.scala
package io.github.graphql.model

import com.fasterxml.jackson.core.`type`.TypeReference

@javax.annotation.Generated(
    value = Array("com.kobylynskyi.graphql.codegen.GraphQLCodegen"),
    date = "2021-03-06T19:48:00+0800"
)
object TopicSuggestionDeclineReasonTO extends Enumeration {

    type TopicSuggestionDeclineReasonTO = Value

    val NOT_RELEVANT: Value = Value("NOT_RELEVANT")
    val PERSONAL_PREFERENCE: Value = Value("PERSONAL_PREFERENCE")
    val TOO_GENERAL: Value = Value("TOO_GENERAL")
    val TOO_SPECIFIC: Value = Value("TOO_SPECIFIC")

}

class TopicSuggestionDeclineReasonTOTypeRefer extends TypeReference[TopicSuggestionDeclineReasonTO.type]

File: src/main/java-graphql/io/github/graphql/java/model/TopicSuggestionDeclineReasonTO.java
package io.github.graphql.java.model;

@javax.annotation.Generated(
    value = "com.kobylynskyi.graphql.codegen.GraphQLCodegen",
    date = "2021-03-07T19:46:14+0800"
)
public enum TopicSuggestionDeclineReasonTO {

    NOT_RELEVANT("NOT_RELEVANT"),
    PERSONAL_PREFERENCE("PERSONAL_PREFERENCE"),
    TOO_GENERAL("TOO_GENERAL"),
    TOO_SPECIFIC("TOO_SPECIFIC");

    private final String graphqlName;

    private TopicSuggestionDeclineReasonTO(String graphqlName) {
        this.graphqlName = graphqlName;
    }

    @Override
    public String toString() {
        return this.graphqlName;
    }

}
